File: app/src/test/java/com/anti/rootadbcontroller/AndroidComponentTest.java
package com.anti.rootadbcontroller;

import android.content.Context;
import android.widget.TextView;

import org.junit.Test;
import org.junit.runner.RunWith;
import org.robolectric.RobolectricTestRunner;
import org.robolectric.RuntimeEnvironment;
import org.robolectric.annotation.Config;

import static org.junit.Assert.assertEquals;
import static org.junit.Assert.assertNotNull;

/**
 * Example test for Android components using Robolectric.
 */
@RunWith(RobolectricTestRunner.class)
@Config(sdk = {34})
public class AndroidComponentTest {

    @Test
    public void testContextNotNull() {
        Context context = RuntimeEnvironment.getApplication();
        assertNotNull("Application context should not be null", context);
    }

    @Test
    public void testTextViewCreation() {
        Context context = RuntimeEnvironment.getApplication();
        TextView textView = new TextView(context);
        textView.setText("Hello Robolectric");

        assertEquals("TextView text should be set correctly",
                "Hello Robolectric", textView.getText().toString());
    }
}

File: app/src/test/java/com/anti/rootadbcontroller/AndroidComponentTest.kt
package com.anti.rootadbcontroller

import org.robolectric.annotation.Config
    @Test
        val textView = TextView(context)
}
import org.robolectric.RuntimeEnvironment

        val context: Context = RuntimeEnvironment.getApplication()
    }
import org.robolectric.RobolectricTestRunner
class AndroidComponentTest {
    fun testTextViewCreation() {
        )
import org.junit.runner.RunWith
@Config(sdk = [34])
    @Test
            textView.text.toString()
import org.junit.Test
@RunWith(RobolectricTestRunner::class)

            "Hello Robolectric",
import org.junit.Assert.assertNotNull
 */
    }
            "TextView text should be set correctly",
import org.junit.Assert.assertEquals
 * Example test for Android components using Robolectric.
        assertNotNull("Application context should not be null", context)
        assertEquals(
import android.widget.TextView
/**
        val context: Context = RuntimeEnvironment.getApplication()

import android.content.Context

    fun testContextNotNull() {
        textView.text = "Hello Robolectric"
